### PR TITLE
Fix for MIN/MAX macros

### PR DESCRIPTION
**Pull request by lucasdemarchi**
See https://github.com/ArduPilot/ardupilot/pull/4907

These patches are backported from upstream.

### CODE REVIEW

File: nuttx/include/stdint.h [hunk 1]
```diff
@@ -55,80 +55,80 @@
...
-#define UINT32_MAX          0xffffffff
+#define INT32_MIN           (-INT32_MAX - 1)
+#define INT32_MAX           2147483647
+#define UINT32_MAX          4294967295u
 
 #ifdef __INT64_DEFINED
-#  define INT64_MIN         0x8000000000000000
-#  define INT64_MAX         0x7fffffffffffffff
-#  define UINT64_MAX        0xffffffffffffffff
+#  define INT64_MIN         (-INT64_MAX - 1ll)
```
Comment (magicrub) [inline nuttx/include/stdint.h]: at first I couldn't figure out why you were subtracting 111. I was very confused. Then I realized it was 1LL :) Perhaps changing that to capital L would make it more readable?

File: nuttx/include/stdint.h [hunk 2]
```diff
@@ -156,46 +156,52 @@
...
+#define UINT32_C(x)         x ## ul
+#define UINT64_C(x)         x ## ull
 #endif
 
 /* Macros for greatest-width integer constant expressions */
 
 #ifdef CONFIG_HAVE_LONG_LONG
-#  define INTMAX_C(x) x ## LL
-#  define UINTMAX_C(x) x ## ULL
+#  define INTMAX_C(x)       x ## ll
```
Comment (magicrub) [inline nuttx/include/stdint.h]: I see you changed it from upper L to lower l. What's the difference?

File: nuttx/include/stdint.h (hunk 1 above)
Comment (lucasdemarchi) [inline nuttx/include/stdint.h]: These are taken from upstream nuttx. I strongly advise against deviating from that. If we want capital letters would be good to do that on upstream.

File: nuttx/include/stdint.h (hunk 2 above)
Comment (lucasdemarchi) [inline nuttx/include/stdint.h]: none. upstream chose to use lowercase.

**Pull request merged**